pkgs/games/sm64pc/default.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv, fetchFromGitHub, requireFile, python3, pkg-config
+, audiofile, SDL2, hexdump, compileFlags ? [], region ? "us" 
+, baseRom ? requireFile {
+    name = "baserom.${region}.z64";
+    message = ''
+      This nix expression requires that baserom.${region}.z64 is
+      already part of the store. To get this file you can dump your Super Mario 64 cartridge's contents
+      and add it to the nix store with nix-store --add-fixed sha256 <FILE>.
+      Note that if you are not using a US baserom, you must overwrite the "region" attribute with either "jp" or "eu".
+    '';
+    sha256 = "17ce077343c6133f8c9f2d6d6d9a4ab62c8cd2aa57c40aea1f490b4c8bb21d91";
+  }
+}:
+
+stdenv.mkDerivation rec {
+  pname = "sm64pc";
+  version = "unstable-2020-05-19";
+
+  src = fetchFromGitHub {
+    owner = "sm64pc";
+    repo = "sm64pc";
+    rev = "c18e70f44eaf628f5795002c29166bbfc4835238";
+    sha256 = "1vjk5g3jkrv21hwv13c7qf1disdv486sx6inn073ijx7g63hyzag";
+  };
+
+  nativeBuildInputs = [ python3 pkg-config ];
+  buildInputs = [ audiofile SDL2 hexdump ];
+
+  inherit compileFlags baseRom;
```

Comment:
You shouldn't need to inherit `compileFlags` if it's not being used in any of the phases. That way you can still use `makeFlags` withing running into infinite recursion.

Reply:
Ah, i thought i had to inherit it when i wanted to use it. I'll remove that.